cluster/src/main/java/com/hartwig/pipeline/metadata/AdditionalApiCalls.java

```diff
@@ -0,0 +1,47 @@
+package com.hartwig.pipeline.metadata;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.hartwig.pipeline.sbpapi.FileResponse;
+import com.hartwig.pipeline.sbpapi.SbpRestApi;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AdditionalApiCalls {
+    private static final AdditionalApiCalls INSTANCE = new AdditionalApiCalls();
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdditionalApiCalls.class);
+    private Map<String, List<ApiFileOperation>> metadata;
+
+    private AdditionalApiCalls() {
+        metadata = new HashMap<>();
+        //String name = RunTag.apply(arguments, metadata.sampleId());
+    }
+
+    public static AdditionalApiCalls instance() {
+        return INSTANCE;
+    }
+
+    public synchronized void register(String filePath, ApiFileOperation extraOperation) {
```

Comment:
another side-effect of singleton. Why isn't register modelled as an Operation somehow?